orbstation/antagonists/traitor/objectives/final_objective/malware_injection.dm

```diff
@@ -0,0 +1,417 @@
+#define APCS_REQUIRED_FOR_MALF 5
+#define INJECTOR_HACK_TIME 20 SECONDS
+
+/datum/traitor_objective/final/malware_injection
+	name = "Hack %NUMBER% APCs, and inject the AI with malware in the %DEPT%"
+	description = "Press the button to call down a drop pod containing a malware injector and a circuit board for a malware injection console. \
+	Use the injector to hack %NUMBER% APCs, and then go to the %DEPT% to build the console. Use the injector on the console to begin \
+	corrupting the AI. WARNING: This process will alert the AI and trigger nearby alarms."
+
+	///Number of APCs that need to be hacked in order to use the injector on the console.
+	var/apcs_required = APCS_REQUIRED_FOR_MALF
+	///Name of the department where the Syndicate AI upload needs to be built.
+	var/department_name
+	///Area type where the Syndicate AI upload needs to be built.
+	var/area/console_area
+	///Checker on whether the pod has been called or not.
+	var/sent_injector = FALSE
+
+/datum/traitor_objective/final/malware_injection/generate_objective(datum/mind/generating_for, list/possible_duplicates)
+	if(!can_take_final_objective())
+		return
+	var/list/active_ais = active_ais() // fetch a list of alive, online AIs
+	for(var/mob/living/silicon/ai in active_ais)
+		if(IS_MALF_AI(ai)) // don't generate this objective if there's already a malf AI
+			return
+	var/list/possible_departments = list(
+		/area/station/medical = "Medbay",
+		/area/station/science = "Science Division",
+		/area/station/engineering = "Engineering Division",
+		/area/station/cargo = "Cargo Bay")
+	var/rand_dept = rand(1, length(possible_departments))
+	var/n = 1
+	for(var/dept_id in possible_departments)
```

Comment:
Can you use `pick(possible_departments )` for this or does it lose one half of the data that way? It feels like there should be an easier way than this loop

Reply:
Yeah this should be a pick().